### ISSUE TITLE
Reject with feedback

### ISSUE THREAD

**Issue by ocean90** [GlotPress/GlotPress#324]
From https://glotpress.trac.wordpress.org/ticket/335

> As an validator I want to decline a validation with a message so the translator know who declined it and why.
> This way I don't need to fix the translation and get's the original translator the credit it deserves.

**Comment by ocean90** [comment]
A possible workflow from #323: https://github.com/GlotPress/GlotPress-WP/issues/323#issuecomment-195285569

**Comment by Presskopp** [comment]
We discussed in our last german #polyglots meeting how the workflow in Glotpress could be improved for PTEs when strings need to be rejected. In result we would explicitly support the idea of 'rejecting with feedback'. There's a mockup on https://github.com/GlotPress/GlotPress-WP/issues/323#issuecomment-195285569. We'd like to suggest the following changes, see image:

![reject](https://user-images.githubusercontent.com/18373563/39253691-b3e8d3dc-48a8-11e8-805c-5ac1d962efc1.png)

We would appreciate checkboxes (including multiple choices) above the 'reject' button with standard reasons as a feedback for the contributor. Also with an extra comment field to give additional explanation, or if another reason is given.
This would be an enormous help for the process of involving people to the translation process.

**Comment by Mte90** [comment]
It is not a bad idea of a preset for feedback but maybe we can do it after the release of the first implementation?

**Comment by La-Geek** [comment]
This is a **must have** feature (with the greatest respect to all your development work, but I would call it a bug, not to have it, that is no claim).

What we (users with global permissions to approve) have at the moment is:

- see an error and reject this string without comment.
- The translator sees a rejected string if he notices it at all.
- Hopefully he'll recognize the mistake, if he doesn't, he'll make the same mistake again.
- And the user with related permissions rejects again (with a little bad luck, we're going around in circles).
- As an option, the user with related permissions tries to contact the translator, but how?
- The only way to contact him is with his nickname.
- With a bit of luck you can reach him by posting in any thread via @nickname.
- Or you're lucky and he's registered in Slack.
- Or you find more informations in his profile, like website.

Can you feel how time-consuming this process is? We have thousands of waiting strings, and in German Polyglots Team less than 10 active PTEs/GTEs with global permissions to approve strings.

In addition: there is no reference in GlotPress to the locale translation team, the rules and where you can find both. Translators often get to Glotpress via the plugin or theme page. Some translators don't even know such a thing exists (but this is another ticket) and is just to show that translators may not even know the local team and the rules to ask their questions.

@Mte90:
I don't know exactly, how the first implementation will look. But on the roadmap you'll see: Glotpress has not reached 3.0 final version and the feature "reject with comment" is planned for 3.2 stable. 

All we ask is to reconsider priorities. We don't need a nice GlotPress, we need one that makes our work easier. We do not know the internal processes and also trust the developers' judgment. But it may not be known how urgently we need this feature.

Thanks to @Presskopp for his detailed description.

**Comment by Mte90** [comment]
As we written on slack, GTE/PTE is not glotpress but WordPress Meta that is another team and has their channels for suggest that.

The first implementation is https://github.com/GlotPress/GlotPress-WP/pull/804 with few screenshots that I made during the development.
We are getting slow for 3.0 that until the release is blocking on develop new feature (this pr has a lot of conflicts that i want to fix only when the new release is available).

**Comment by La-Geek** [comment]
We are asking for a feature in GlotPress, nothing more. And this seems to be the right place for this. Sorry, I don't understand, I will ask you in Slack what you mean. 

Thank you for clarification at Slack. We are looking forward to the further steps in the development :)

**Comment by Presskopp** [comment]
didn't notice https://github.com/GlotPress/GlotPress-WP/pull/804 - great to see there's already a history of progress on that

**Comment by davidperezgar** [comment]
Yes! It's a must feature!

**Comment by nabsul** [comment]
Hi, it seems this feature has been under development for a very long time. Is there still work being done? Or has all that work been scrapped?

I'm asking because I would like to develop and contribute this feature if it's not being worked on.

**Comment by Mte90** [comment]
The idea at WCEU 2019 at glotpress table was to remove the support from GP of the feedback because has some issues like missing notifications or more features.
Also because on wp.org there was the idea to integrate with the forum itself because has everything instead of a new feature from scratch that already exists.

**Comment by nabsul** [comment]
Just to make sure I'm understanding everything:

> remove the support from GP ... because .... missing notifications or more features

This is really interesting to me. Personally I'd rather have a partial feature rather than none at all.

> there was the idea to integrate with the forum

Is there any work already being done on this? Also, would this approach mean that users would have to install some forum plugin to have this feature?

**Comment by Mte90** [comment]
No we are not working so much on GP as you can see is like dead also on slack.
The point about the missing feature was that as it is now is very incomplete, as example someone leave a feedback the author of the localization need a notification he cannot check all the strings to see if there are.
So also because in WP right there is the plan to do https://make.wordpress.org/core/2020/01/14/wp-notify-meeting-recap-january-13-2020/ but is not avalaible yet.

**Comment by nabsul** [comment]
Thank you for the background information. This is helpful. But I'm still lost regarding next steps for me personally (and my friend @ramico who is also interested). 

We want to work on this feature if the resulting work has the chance of being shipped into GlotPress and translate.wordpress.org. We both spend time translating there and this feature would help us work more efficiently.

I feel like we could develop an initial solution that solves most of the frustrations in this area. It's also possible to write the code in a general way that plugging in notifications will be easy when that feature is developed. 

So far I'm interpreting your previous comments as "There are too many unknowns and dependencies, so working on this feature is not possible.". Am I understanding this correctly, or are there options to proceed on this?  

Also a more specific question:

> So also because in WP right there is the plan to do https://make.wordpress.org/core/2020/01/14/wp-notify-meeting-recap-january-13-2020/ but is not avalaible yet.

If the idea is to integrate with a forum (bbPress I assume?), wouldn't notifications be solved by the forum software so we wouldn't have to worry about it?

**Comment by Mte90** [comment]
Yes the idea was to leave all the notifications and feedback system, instead as it is now https://github.com/GlotPress/GlotPress-WP/pull/975 to use at all the forum so all the features are already available without need to develop anything.

The plan was to remove all the code of that feature and move it in a new plugin but probably the whole feature can be a plugin that integrate bbpress inside glotpress.
I suggest to you to do a plugin and later evaluate if integrate it in GP so the development is not stopped from the actual status of GP but I invite you to join the #glotpress channel on the wordpress slack for other discussion (so we can revive it) :-)

**Comment by psmits1567** [comment]
This issue can be closed as we now have the discussion system where you can reject and give comment on it
